File: training-data/python/sieve_of_eratosthenes.py
def sieve(limit: int):
    is_prime = [True] * (limit + 1)
    is_prime[0:2] = [False, False]
    for n in range(2, int(limit ** 0.5) + 1):
        if is_prime[n]:
            for multiple in range(n * n, limit + 1, n):
                is_prime[multiple] = False
    return [n for n, prime in enumerate(is_prime) if prime]


if __name__ == "__main__":
    print(sieve(50))

File: training-data/dart/sieve_of_eratosthenes.dart
List<int> sieve(int limit) {
  final isPrime = List<bool>.filled(limit + 1, true);
  isPrime[0] = false;
  if (limit >= 1) isPrime[1] = false;
  for (var n = 2; n * n <= limit; n++) {
    if (isPrime[n]) {
      for (var m = n * n; m <= limit; m += n) {
        isPrime[m] = false;
      }
    }
  }
  return [for (var n = 0; n <= limit; n++) if (isPrime[n]) n];
}

void main() {
  print(sieve(50));
}
